File: backend/src/main/java/me/performancereservation/domain/reservation/dto/ReservationPageResponse.java
package me.performancereservation.domain.reservation.dto;

import me.performancereservation.domain.reservation.enums.ReservationStatus;

import java.time.LocalDateTime;

/**
 * 예약 목록 조회용 응답 Dto
 */
public record ReservationPageResponse(
        // Reservation
        Long reservationId, // 예약 ID
        int quantity, // 티켓 수량
        ReservationStatus status, // 예약 상태
        LocalDateTime createdAt, // 예약 일시

        // Performance
        String title, // 공역 제목
        String venue, // 공연 장소
        int ticketPrice, // 티켓 가격

        // Calculated value
        int totalPrice // 총 결제 금액
) {}

File: backend/src/main/kotlin/me/performancereservation/domain/reservation/dto/ReservationPageResponse.kt
package me.performancereservation.domain.reservation.dto

import me.performancereservation.domain.reservation.enums.ReservationStatus
import java.time.LocalDateTime

/**
 * 예약 목록 조회용 응답 Dto
 */
data class ReservationPageResponse(
    // Reservation
    val reservationId: Long,  // 예약 ID
    val quantity: Int,  // 티켓 수량
    val status: ReservationStatus,  // 예약 상태
    val createdAt: LocalDateTime,  // 예약 일시

    // Performance
    val title: String,  // 공역 제목
    val venue: String,  // 공연 장소
    val ticketPrice: Int,  // 티켓 가격

    // Calculated value
    val totalPrice: Int // 총 결제 금액
)
